apps/meteor/ee/server/models/raw/WorkspaceCredentials.ts

```diff
@@ -0,0 +1,68 @@
+import type { RocketChatRecordDeleted, IWorkspaceCredentials } from '@rocket.chat/core-typings';
+import type { IWorkspaceCredentialsModel } from '@rocket.chat/model-typings';
+import type { Collection, Db, DeleteResult, Filter, IndexDescription, UpdateResult } from 'mongodb';
+
+import { BaseRaw } from '../../../../server/models/raw/BaseRaw';
+
+export class WorkspaceCredentialsRaw extends BaseRaw<IWorkspaceCredentials> implements IWorkspaceCredentialsModel {
+	constructor(db: Db, trash?: Collection<RocketChatRecordDeleted<IWorkspaceCredentials>>) {
+		super(db, 'workspace_credentials', trash);
+	}
+
+	protected modelIndexes(): IndexDescription[] {
+		return [{ key: { _id: 1, scopes: 1, expirationDate: 1, accessToken: 1 }, unique: true }];
```

Comment:
Since _id is already unique, you would need an actual duplicated document for this index to trigger the unique warning :/

If you remove the _id from it, then it would properly validate that scopes, expiration and access token should be unique on any document